src/large.c

```diff
@@ -291,32 +290,39 @@ large_ralloc(tsdn_t *tsdn, arena_t *arena, extent_t *extent, size_t usize,
  * independent of these considerations.
  */
 static void
-large_dalloc_impl(tsdn_t *tsdn, extent_t *extent, bool junked_locked) {
-	arena_t *arena;
-
-	arena = extent_arena_get(extent);
+large_dalloc_prep_impl(tsdn_t *tsdn, arena_t *arena, extent_t *extent,
+    bool junked_locked) {
 	malloc_mutex_lock(tsdn, &arena->large_mtx);
-	ql_remove(&arena->large, extent, ql_link);
+	extent_list_remove(&arena->large, extent);
 	malloc_mutex_unlock(tsdn, &arena->large_mtx);
 	if (!junked_locked) {
 		large_dalloc_maybe_junk(extent_addr_get(extent),
 		    extent_usize_get(extent));
 	}
-	arena_extent_dalloc_large(tsdn, arena, extent, junked_locked);
+	arena_extent_dalloc_large_prep(tsdn, arena, extent, junked_locked);
```

Comment:
Hmm. I think I have seen this a couple of times. I'll see if I can report bug somewhere.